### PR TITLE
chore(docs): update v2 migration docs

### PR DESCRIPTION
**Pull request by kennylam**
### Description

Update v2 migration docs to be a bit more consistent in wording, formatting, etc. This is not final, just a bit of a clean up.

### Changelog

**New**

- {{new thing}}

**Changed**

- {{changed thing}}

**Removed**

- {{removed thing}}

### CODE REVIEW

File: docs/dotcom-v2-migration.md [hunk 1]
```diff
@@ -1,125 +1,124 @@
...
 
-## List of component changes
+## Components
 
 | Component                 | Changes                                                                                                                                        |
 | ------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------- |
 | back-to-top               | No API changes.                                                                                                                                |
 | background-media          | No API changes.                                                                                                                                |
-| button                    | This component is deprecated in v2 in favor for `button-expressive`.                                                                           |
+| button                    | Deprecated in favor of `button-expressive`.                                                                           |
```
Comment (ariellalgilmore) [inline docs/dotcom-v2-migration.md:19]: didn't we end up actually keeping button?

**Review by ariellalgilmore — APPROVED** [review]
Small comment, but LGTM otherwise!

File: docs/dotcom-v2-migration.md (hunk 1 above)
Comment (kennylam) [reply docs/dotcom-v2-migration.md:19]: I didn't work on this so I'm not 100% sure. But I just spoke with @oliviaflory about it and at the very least it seems like the components are swapped (it should be `button-expressive` deprecated in favor of `button`).

Comment (ariellalgilmore) [reply docs/dotcom-v2-migration.md:19]: ahh yup: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/blob/feat/carbon-for-ibm-dotcom-v2/packages/web-components/src/components/button/button-expressive.ts#L252

Comment (kennylam) [reply docs/dotcom-v2-migration.md:19]: Ok, got it – `button-expressive` is getting deprecated. Instead, all button usage should come from `@carbon/web-components`.

Comment (kennylam) [reply docs/dotcom-v2-migration.md]: Right.

**Pull request merged**